docs/BYO_INFRASTRUCTURE.adoc

```diff
@@ -0,0 +1,60 @@
+= Using the CASL Tools with Static Infrastructure (Bring Your Own Infrastructure)
+
+This guide walks through the process of using the tools in this repository to automate the stand up of OpenShift clusters on Infrastructure where there is currently no automated provisioner. This process can easily be used in on-premise installs where the infrastructure is not currently API driven and has yet to be automated in a way that's compatible with CASL.
+
+== What Infrastructure do I need before I run CASL?
+
+If you read through the link:PROVISIONING_ARCH.md[CASL Provisioning Architecture documentation], you'll notice the following description of our _Provisioning Phase_:
+
+_Provision Infrastructure_ creates the servers/instances and any other supporting infrastructure required (network, DNS, load balancers, etc.). In general this phase
+  - Begins with having credentials with proper access to an Infrastructure provider.
+  - Ends with servers provisioned and running with an OS installed and SSH keys (with root or sudo) pushed.
+
+In many on-premise installs, the infrastructure may not yet be ready to automate to the level that our provisioning architecture expects. That's fine, we can still use the CASL Tools to automate the other 3 phases of install automation:
+
+2. _Pre Install_
+3. _Install_
+4. _Post Install_
+
+The link:../inventory/sample.byo.example.com.d/[sample.byo.example.com] inventory in this repo represents an ansible inventory that can be used in such a scenario.
+
+We recommend giving the link:http://v1.uncontained.io/playbooks/installation/#building-the-infrastructure[Installing a Highly Available OpenShift Cluster] guide a read for details on what infrastructure to build out. Specifically, the link:http://v1.uncontained.io/playbooks/installation/#cluster-design-architecture[Cluster Design & Architecture Section] lays out all of the design decisions that need to be made when preparing a cluster, and the link:http://v1.uncontained.io/playbooks/installation/#building-the-infrastructure[Building the Infrastructure] section walks through the components that should be built as part of a manual _provisioning_ step.
+
+== Prepping the Inventory
+
+While the link:../inventory/sample.byo.example.com.d/[sample.byo.example.com] sample inventory provides most of the variables you need to get your cluster prepped and installed, there are a few variables in the `sample.byo.example.com.d/inventory/group_vars/all.yml` file that need to be filled in to match your environment.
+
+- The `ansible_user: <your ssh key user>` variable must match the user you will use to log into the cluster machines
+- You must fill in the `rhsm_` variables to match your Red Hat Subscription Manager account. You can do this one of two ways:
```

Comment:
May want to mention that `rhsm_register: True` has to be set to successfully register the hosts.